lib/Optimizer/GraphOptimizer.cpp

```diff
@@ -205,6 +205,39 @@ static Node *simplifyNode(Node *node, Function *F) {
   return node;
 }
 
+// 2D vector of dimentions mapping a dimention that is reshaped into either 1
+// or 2 dimentions by the first reshape node in a ChannelShuffle operation
+using ChannelShuffleDimGroups =
+    llvm::SmallVector<llvm::SmallVector<unsigned_t, 2>, max_tensor_dimensions>;
+
+/// \returns the dims of \p node grouped by which input dimention they came from
+static bool groupDimentionsFromChannelShuffleReshape(
+    const ReshapeNode &node, ChannelShuffleDimGroups &dimGroups) {
+  const auto dims = node.getDims();
+  const auto inputDims = node.getInput().dims();
+  if (dims.size() != inputDims.size() + 1) {
+    return false;
+  }
+  dimGroups.resize(inputDims.size());
+  bool foundReshapedDim = false;
+  int dimIdx, groupIdx;
+  for (dimIdx = 0, groupIdx = 0; dimIdx < dims.size(); ++dimIdx, ++groupIdx) {
```

Comment:
I think you should be able to do 
```suggestion
  for (int dimIdx = 0, groupIdx = 0; dimIdx < dims.size(); ++dimIdx, ++groupIdx) {
```
And remove `int dimIdx, groupIdx;` above. And actually it should be `size_t`, not `int`